src/network/mcpe/cache/CraftingDataCache.php

```diff
@@ -141,6 +145,33 @@ private function buildCraftingDataCache(CraftingManager $manager) : CraftingData
 			}
 		}
 
+		$index ??= 0;
```

Comment:
This looks suspicious.

Reply:
it's there to prevent errors in case no crafting recipe gets registered. the indexes are incrementing and are not reseted when switching from crafting to smithing